pkg/test/ginkgo/options_monitor_events.go

```diff
@@ -113,6 +117,60 @@ func (o *MonitorEventsOptions) End(ctx context.Context, restConfig *rest.Config,
 	o.endTime = &t
 	o.recordedResources = o.monitor.CurrentResourceState()
 
+	removeNTimes := regexp.MustCompile(`\s+\(\d+ times\)`)
+
+	currentLogLevel := logrus.GetLevel()
+	logrus.SetLevel(logrus.DebugLevel)
+
+	// Scan the unsorted events looking for those marked only as pathological/true.
+	logrus.Debug("Looking through unsorted events: ", len(o.monitor.GetUnsortedEvents()))
+	for _, pathologicalEvent := range o.monitor.GetUnsortedEvents() {
+		if !strings.Contains(pathologicalEvent.Message, duplicateevents.PathologicalMark) {
+			// We only are interested in those EventIntervals with pathological/true
+			continue
+		}
+		if strings.Contains(pathologicalEvent.Message, duplicateevents.InterestingMark) {
+			// If this message is known, we don't need to process it
+			continue
+		}
+		// If we get here, pathologicalEvent.Message has pathological/true which means it happened above threshold
+		// and is not known.  Search back for any matching EventIntervals that are less than threshold.
+
+		// EventIntervals marked with pathological/true alone have a hmsg/HASH on the end of the locator
+		// and happened more than threshold times.
+		pathoMsgWithoutTimes := removeNTimes.ReplaceAllString(pathologicalEvent.Message, "")
+		hash := sha256.Sum256([]byte(pathoMsgWithoutTimes))
+		hashStr := fmt.Sprintf("%x", hash)[:10]
+
+		logrus.Debugf("Looking at previous events for %s, msg=%s\n", pathologicalEvent.Locator, pathologicalEvent.Message)
+		logrus.Debugf("  %s - %s\n", pathologicalEvent.From.Format("15:04:05"), pathologicalEvent.To.Format("15:04:05"))
+		matchCount := 0
+		unsortedEventsToScan := o.monitor.GetUnsortedEvents()
+		for i := range unsortedEventsToScan {
```

Comment:
Actually we can probably do better.

- declare a slice of messages to look (or maybe a StringSet from the kube library) for with the n times removed
- loop through the events var *backwards*, they should already be sorted via m.Intervals() function
  - if we see a n times unknown pathological event, add it's message to the slice
  - otherwise if the message matches one in our slice, it's a match

Then you just have to go through once. It should also eliminate the duplicates you mentioned in your PR comment.